php/WC_Gateway_Blockonomics.php

```diff
@@ -172,64 +134,106 @@ public function check_blockonomics_callback()
+        $order_id = isset($_REQUEST["finish_order"]) ? $_REQUEST["finish_order"] : "";
+        if ($order_id) {
+            $order = $orders[$order_id];
+            $wc_order = new WC_Order($order_id);
             echo $order['order_id'];
             wp_redirect($wc_order->get_checkout_order_received_url());
             exit();
         }
-        $address = isset($_REQUEST['get_order']) ? $_REQUEST['get_order'] : "";
-        if ($address) {
+        $order_id = isset($_REQUEST['get_order']) ? $_REQUEST['get_order'] : "";
+        $crypto= isset($_REQUEST['crypto']) ? $_REQUEST['crypto'] : "";
+        if ($order_id && $crypto) {
+            include_once 'Blockonomics.php';
+            $blockonomics = new Blockonomics;
+            global $woocommerce;
+            $order = $orders[$order_id];
+            $wc_order = new WC_Order($order_id);
+            if(get_woocommerce_currency() != 'BTC'){
+                $price = $blockonomics->get_price(get_woocommerce_currency(), $crypto);
+                $price = $price * 100/(100+get_option('blockonomics_margin', 0));
+            }else{
+                $price = 1;
+            }
+            $currentAddress = get_post_meta($order_id,$crypto .'_address');
+            if(isset($currentAddress[0])) {
+                $address = $currentAddress[0];
+            } else {
+                $responseObj = $blockonomics->new_address(get_option("blockonomics_callback_secret"), $crypto);
+                if($responseObj->response_code != 200) {
+                    return;
+                }
+                $address = $responseObj->address;
+            }
+
+            $order = array(
+                'value'              => $wc_order->get_total(),
+                'currency'           => get_woocommerce_currency(),
+                'order_id'           => $order_id,
+                'status'             => -1,
+                'address'            => $address,
+                'satoshi'            => intval(round(1.0e8*$wc_order->get_total()/$price)),
+                'crypto'             => $crypto,
+                'timestamp'          => time()
+            );
+
+            $orders[$order_id] = $order;
+            update_option('blockonomics_orders', $orders);
+            update_post_meta($wc_order->get_id(), ''. $order['crypto'] .'_address', $order['address']);
+
             header("Content-Type: application/json");
-            exit(json_encode($orders[$address]));
+            exit(json_encode($orders[$order_id]));
         }
 
         $callback_secret = get_option("blockonomics_callback_secret");
         $secret = isset($_REQUEST['secret']) ? $_REQUEST['secret'] : "";
         $network_confirmations=get_option("blockonomics_network_confirmation",2);
         if ($callback_secret  && $callback_secret == $secret) {
             $addr = isset($_REQUEST['addr']) ? $_REQUEST['addr'] : "";
-            if (array_key_exists($addr, $orders)){
```

Comment:
This change is doing same thing as previous. Its cleaner to use `array_key_exists()` than to implement the same thing in a `for each` loop